editor/src/components/canvas/canvas-strategies/absolute-reparent-to-flex-strategy.tsx

```diff
@@ -0,0 +1,248 @@
+import * as EP from '../../../core/shared/element-path'
+import * as PP from '../../../core/shared/property-path'
+import { AllElementProps } from '../../editor/store/editor-state'
+import { ElementInstanceMetadataMap } from '../../../core/shared/element-template'
+import {
+  CanvasStrategy,
+  emptyStrategyApplicationResult,
+  InteractionCanvasState,
+  StrategyApplicationResult,
+} from './canvas-strategy-types'
+import { InteractionSession, StrategyState } from './interaction-state'
+import { ParentBounds } from '../controls/parent-bounds'
+import { ParentOutlines } from '../controls/parent-outlines'
+import { DragOutlineControl } from '../controls/select-mode/drag-outline-control'
+import { MetadataUtils } from '../../../core/model/element-metadata-utils'
+import { getDragTargets } from './shared-absolute-move-strategy-helpers'
+import { getReparentTarget } from '../canvas-utils'
+import { ElementPath, PropertyPath } from '../../../core/shared/project-file-types'
+import { reparentElement } from '../commands/reparent-element-command'
+import { getReorderIndex } from './flex-reorder-strategy'
+import { offsetPoint } from '../../../core/shared/math-utils'
+import { reorderElement } from '../commands/reorder-element-command'
+import { CSSCursor } from '../canvas-types'
+import { setCursorCommand } from '../commands/set-cursor-command'
+import { setElementsToRerenderCommand } from '../commands/set-elements-to-rerender-command'
+import { updateHighlightedViews } from '../commands/update-highlighted-views-command'
+import { CanvasCommand, foldAndApplyCommandsInner } from '../commands/commands'
+import { deleteProperties } from '../commands/delete-properties-command'
+import { updateSelectedViews } from '../commands/update-selected-views-command'
+
+function reparentTargetFromInteractionSession(
+  filteredSelectedElements: Array<ElementPath>,
+  interactionSession: InteractionSession,
+  canvasState: InteractionCanvasState,
+  strategyState: StrategyState,
+): {
+  shouldReparent: boolean
+  newParent: ElementPath | null
+  shouldReorder: boolean
+} {
+  const reparentResult = getReparentTarget(
+    filteredSelectedElements,
+    filteredSelectedElements,
+    interactionSession.metadata,
+    [],
+    canvasState.scale,
+    canvasState.canvasOffset,
+    canvasState.projectContents,
+    canvasState.openFile,
+    strategyState.startingAllElementProps,
+  )
+  if (reparentResult.newParent == null) {
+    return {
+      ...reparentResult,
+      shouldReorder: false,
+    }
+  } else {
+    // The target is in a flex container.
```

Comment:
so if the found target parent is _in_ a flex container, then we return the target parent's parent as the actual target parent? 
could you extend the comment here with a brief explanation? I _think_ we will have to change the code in `getReparentTarget` as we move forward with this work